app/lib/features/space/widgets/edit_space_sheet.dart

```diff
@@ -190,96 +165,51 @@ class _EditSpacePageConsumerState extends ConsumerState<EditSpacePage> {
+          ? null
+          : () async {
+              // check permissions before updating space
+              bool havePermission = await permissionCheck();
+              if (!havePermission) {
+                // ignore: use_build_context_synchronously
+                customMsgSnackbar(
+                  context,
+                  'Cannot edit space with no permissions',
+                );
+              }
+              if (context.mounted) {
+                showAdaptiveDialog(
+                  barrierDismissible: false,
+                  context: context,
+                  builder: (context) => DefaultDialog(
+                    title: Text(
+                      'Updating Space',
+                      style: Theme.of(context).textTheme.titleSmall,
                     ),
+                    isLoader: true,
                   ),
-                  subtitle: const Text('Cannot edit space with no permissions'),
-                  actions: <Widget>[
-                    DefaultButton(
-                      onPressed: () => context.pop(),
-                      title: 'Okay',
-                      style: ButtonStyle(
-                        backgroundColor: MaterialStateProperty.all<Color>(
-                          Theme.of(context).colorScheme.success,
-                        ),
-                      ),
-                    ),
-                  ],
-                ),
-              );
-              return;
-            }
-            final roomId = await _handleUpdateSpace(context);
-            debugPrint('Space Updated: $roomId');
-            // We are doing as expected, but the lints triggers.
-            // ignore: use_build_context_synchronously
-            if (!context.mounted) {
-              return;
-            }
-            context.pushNamed(
-              Routes.space.name,
-              pathParameters: {
-                'spaceId': roomId.toString(),
-              },
-            );
-          },
-          style: ElevatedButton.styleFrom(
-            backgroundColor: titleInput.isNotEmpty
-                ? Theme.of(context).colorScheme.success
-                : Theme.of(context).colorScheme.success.withOpacity(0.6),
-            shape: RoundedRectangleBorder(
-              borderRadius: BorderRadius.circular(6),
-            ),
-            foregroundColor: Theme.of(context).colorScheme.neutral6,
-            textStyle: Theme.of(context).textTheme.bodySmall,
-          ),
-          child: const Text('Save changes'),
-        ),
-      ],
+                );
+                final roomId = await _handleUpdateSpace(context);
+                debugPrint('Space Updated: $roomId');
+                // We are doing as expected, but the lints triggers.
+                // ignore: use_build_context_synchronously
+                if (!context.mounted) {
+                  return;
+                }
+                context.pop();
+                context.pop();
```

Comment:
we need a better handling of this popping... it is hard to follow...

```suggestion
                context.pop(); // pop the loading screen
                context.pop(); // pop the edit sheet
```